hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcClient.java

```diff
@@ -116,10 +118,13 @@
   /** Optional perf monitor. */
   private final RouterRpcMonitor rpcMonitor;
 
+  private final Configuration clientConf;
+
   /** Pattern to parse a stack trace line. */
   private static final Pattern STACK_TRACE_PATTERN =
       Pattern.compile("\\tat (.*)\\.(.*)\\((.*):(\\d*)\\)");
 
+  private static final String CLIENT_IP_STR = "clientIp";
```

Comment:
Is there something else we need to add?

Reply:
Now don't know Whether add anything else except client ip, any thoughts?

Reply:
One of the things that was brought up in the past was to send back in the context which was the namespace that executed this.
Anyway, let's focus on the locality one for now.
Does it make sense to add a test to make sure we can do proper locality from the Namenode side with this change? Or should we leave this for the future.

Reply:
[HDFS-13248](https://issues.apache.org/jira/browse/HDFS-13248) focuses on the locality, here we focus on audit log, does it make sense?

Reply:
Makes sense, let's just keep in mind HDFS-13248 when doing this.
So far it looks like is covered.